src/Microsoft.IdentityModel.Tokens/BaseConfigurationManager.cs

```diff
@@ -76,9 +93,39 @@ public BaseConfiguration LastKnownGoodConfiguration
             {
                 _lastKnownGoodConfiguration = value ?? throw LogHelper.LogArgumentNullException(nameof(value));
                 _lastKnownGoodConfigFirstUse = DateTime.UtcNow;
+
+                if (_lkgConfigurationCache == null)
+                    _lkgConfigurationCache = new Dictionary<BaseConfiguration, DateTime>(BaseConfigurationComparer);
+                
+                _lkgConfigurationCache[_lastKnownGoodConfiguration] = DateTime.UtcNow + LastKnownGoodLifetime;
+
+                //remove expired configuration to avoid memory leak 
+                foreach (KeyValuePair<BaseConfiguration, DateTime> lkgConfiguration in _lkgConfigurationCache)
```

Comment:
>foreach [](http://example.com/codeflow?start=16&length=7)

Does this work? My memory of modifying a collection while iterating over it was that it will throw an exception.

[https://stackoverflow.com/questions/2347269/modifying-net-dictionary-while-enumerating-through-it](https://stackoverflow.com/questions/2347269/modifying-net-dictionary-while-enumerating-through-it)

Generally you'll want to create a collection and remove once finished with iteration.